File: ejercicios hechos/Python/estructuras alternativas/act16.py
#Programa: act16.py
#Propósito:La política de cobro de una compañía telefónica es: cuando se realiza una llamada, el cobro es por el tiempo que ésta dura, de tal forma que los primeros cinco minutos cuestan 1 euro, los siguientes tres, 80 céntimos, los siguientes dos minutos, 70 céntimos, y a partir del décimo minuto, 50 céntimos. Además, se carga un impuesto de 3 % cuando es domingo, y si es otro día, en turno de mañana, 15 %, y en turno de tarde, 10 %. Realice un algoritmo para determinar cuánto debe pagar por cada concepto una persona que realiza una llamada.
#Autor: Jose Manuel Serrano Palomo.
#Fecha: 17/10/2019
#
#Variables a usar:
# precio_llamada será el precio total de la llamada
# día es el día en el que llama y que determinará el impuesto
# tiempo es lo que dura la llamada
# turno en el caso de que no sea el domingo nos dirá un impuesto diferente segun si es te tarde o mañana
#
#Algoritmo:
# si el tiempo es igual o inferior a 5 mins precio_llamada <-- 1 eur
# si el tiempo es igual o inferior a 8 mins precio_llamada <-- 1 eur + 0.80 cents
# si el tiempo es igual o inferior a 10 mins precio_llamada <-- 1 eur + 0.80 cents + 0.70 cents
# si el tiempo es superior a 10 mins precio_llamada <-- 1 eur + 0.80 cents + 0.70 cents + 0.50 cents
# si fue un domingo 3% de impuestos añadidos a precio_llamada
# en caso contrario LEER si fue turno de mañana 15% añadido o tarde 10% añadido
# ESCRIBIR el resultado

print("calcular el precio de una llamada telefónica")
print("--------------------------------------------\n")

#Leemos los datos
dia = str(input("Introduce el día que realiza la llamada: "))
tiempo = int(input("Introduce el tiempo que dura la llamada en minutos: "))

#Realizamos las comprobaciones
if tiempo <= 5:
    precio_llamada = 1
    
else:
    if tiempo <= 8:
        precio_llamada = 1 + 0.80
    
    else:
        if tiempo <= 10:
            precio_llamada = 1 + 0.80 + 0.70
        
        else: 
            if tiempo > 10:
                precio_llamada = 1 + 0.80 + 0.70 + 0.50
                
            else:
                print("los datos son erroneos introduce un numero entero")

if dia == "domingo" or dia == "Domingo" or dia == "DOMINGO":
    precio_llamada = round(precio_llamada * 1.03, 2)
    print(f"El precio de la llamada será {precio_llamada}")

else:
    if dia != "domingo" or dia != "Domingo" or dia != "DOMINGO":
        turno = str(input("Introduce en que turno se realiza la llamada : "))
    
    else:
        print("los datos son erroneos introduce un día de la semana")
    
    if turno == "mañana":
        precio_llamada = round(precio_llamada * 1.15,2)
        print(f"El precio de la llamada será {precio_llamada}")
        
    else:
        if turno == "tarde":
            precio_llamada = round(precio_llamada * 1.10,2)
            print(f"El precio de la llamada será {precio_llamada}")
        
        else:
            print("los datos son erroneos introduce turno de tarde o mañana")

File: ejercicios hechos/Java/estructuras alternativas/act16.java
/*Programa: act16.java
 * 
 *Propósito: La política de cobro de una compañía telefónica es: cuando se realiza una llamada, el cobro es por el tiempo que ésta dura, de tal forma que los primeros cinco minutos cuestan 1 euro, los siguientes tres, 80 céntimos, los siguientes dos minutos, 70 céntimos, y a partir del décimo minuto, 50 céntimos. Además, se carga un impuesto de 3 % cuando es domingo, y si es otro día, en turno de mañana, 15 %, y en turno de tarde, 10 %. Realice un algoritmo para determinar cuánto debe pagar por cada concepto una persona que realiza una llamada.
 * 
 *Autor: Jose Manuel Serrano Palomo.
 * 
 *Fecha: 09/11/2019
 *
 *Análisis:
 * El precio de una llama se calcula dependiendo del tiempo
 * Cada ciertos minutos el precio disminye hasta quedare en los 10 mins 
 * y dependiendo del dia y el horario tambien varía el precio,
 *
 *Diseño:
 * 1. Leer tiempo
 * 2. Leer si la llamada es en domingo
 * 3. Si no es en domingo, leer el turno (Mañana o Tarde)
 * 4. Si tiempo <5 coste=tiempo*100
 * 5. Si tiempo<8 coste=(tiempo-5)*80 + 500 (el coste de los cinco primeros minutos)
 * 6. Si tiempo<10 coste=(tiempo-8)*70 + 240 (el coste desde el minuto 6 al 8) + 500 (el coste de los cinco primeros minutos)
 * 7. Si tiempo>10 coste=(tiempo-10)*50 + 140 (el coste desde el minuto 9 al 10) + 240 (el coste desde el minuto 6 al 8) + 500 (el coste de los cinco primeros minutos)
 * 8. Si la llamada es en domingo coste = coste + 3%
 * 9. Si la llamada es otro día por la mañana coste = coste + 15%
 * 10. Si la llamada es otro día por la mañana coste = coste + 10%
 * 11. Mostrar coste final en euros
*/

import java.util.Scanner;
public class act16 {
    public static void main(String[] args) {
        System.out.println("Programa para calcular el precio de una llamada");
        System.out.println("-----------------------------------------------");
        Scanner s = new Scanner(System.in);
        double coste; // centimos
        String esDomingo;
        int tiempo;
        String turno="";

        System.out.print("¿Cuánto tiempo es la llamada?: ");
        tiempo = s.nextInt(); s.nextLine();
        System.out.print("¿Es Domingo? (S/N): ");
        esDomingo = s.nextLine();
        if (esDomingo.toUpperCase().equals("N")) {
          System.out.print("¿Qué turno: Mañana o Tarde? (M/T)?: ");
          turno = s.nextLine();
        }

        if (tiempo<=5) {
          coste = tiempo*100;
        } else if (tiempo<=8) {
          coste = (tiempo-5)*80+500;
        } else if (tiempo<=10) {
          coste = (tiempo-8)*70+240+500;
        } else {
          coste = (tiempo-10)*50+140+240+500;
        }
        if (esDomingo.toUpperCase().equals("S")) {
          coste += coste*0.03;
        } else if (turno.toUpperCase().equals("M")) {
          coste += coste*0.15;
        } else {
          coste += coste*0.10;
        }
        System.out.println("El coste de la llamada: "+coste/100+" euros.");
    }
}
